### DISCUSSION TITLE
How to prevent session flash message caching?

### DISCUSSION THREAD

**Discussion by rrdirnens** [spatie/laravel-responsecache#410] [Q&A]
The title, pretty much. 
Laravel has the option to [redirect with flashed session data](https://laravel.com/docs/9.x/responses#redirecting-with-flashed-session-data), but this also gets cached. Is there a way to avoid this?

**Comment by lianmafutra** [comment]
you can use custom replacer in config/responsecache.php

  'replacers' => [
        \Spatie\ResponseCache\Replacers\CsrfTokenReplacer::class,
        ReplacerCache::class, 
    ],


// example class ReplacerCache to handle session reset with replace string your code session code
 // ' if (true) {'  replace with   ' if (false) {',


<?php

namespace App\Utils;

use Spatie\ResponseCache\Replacers\Replacer;
use Symfony\Component\HttpFoundation\Response;

class ReplacerCache implements Replacer
{

   public function prepareResponseToCache(Response $response): void
   {
      if (!$response->getContent()) {
         return;
      }

      $response->setContent(str_replace(
         ' if (true) {',
         ' if (false) {',
         $response->getContent()
      ));
   }

   public function replaceInCachedResponse(Response $response): void
   {
      if (!$response->getContent()) {
         return;
      }
   }

}

**Comment by DannyCooper** [comment]
I'm also having this issue. Did you find a solution @rrdirnens?

**Reply by lianmafutra** [reply]
https://github.com/spatie/laravel-responsecache/discussions/410#discussioncomment-6502921

**Reply by DannyCooper** [reply]
Thank you @lianmafutra, but has some code been lost in the formatting? I can't see how that applies to flash message/sessions.

**Comment by beardcoder** [comment]
A Simple workaround is a custom Middleware that extend the Cache Middleware

```php
<?php

namespace App\Http\Middleware;

use Closure;
use Illuminate\Http\Request;
use Spatie\ResponseCache\Middlewares\CacheResponse;
use Symfony\Component\HttpFoundation\Response;

class CheckQueryForCaching extends CacheResponse
{
  public function handle(Request $request, Closure $next, ...$args): Response
  {
    if ($request->getQueryString()) {
      return $next($request);
    }

    return parent::handle($request, $next, $args);
  }
}
```
